src/device/device-management.ts

```diff
@@ -70,28 +103,67 @@ export async function createCameraTrack(
  * @returns A LocalTrack object or an error.
  */
 export async function createMicrophoneTrack(
-  constraints?: AudioDeviceConstraints
+  constraints?: MicrophoneConstraints
 ): Promise<LocalMicrophoneTrack> {
   let stream: MediaStream;
   try {
-    stream = await media.getUserMedia({ audio: { ...constraints } });
+    stream = await media.getUserMedia({ audio: generateAudioConstraints(constraints) });
   } catch (error) {
-    throw new WcmeError(
+    throw new WebrtcError(
       ErrorTypes.CREATE_MICROPHONE_TRACK_FAILED,
       `Failed to create microphone track ${error}`
     );
   }
-  return new LocalMicrophoneTrack(stream);
+  // See if we can just pass the track and not streams
+  return new LocalMicrophoneTrack(stream.getAudioTracks()[0]);
 }
 
 /**
  * Creates a display video track.
  *
+ * @param constraints - Display constraints for screen sharing.
  * @returns A Promise that resolves to a LocalDisplayTrack.
  */
-export async function createDisplayTrack(): Promise<LocalDisplayTrack> {
-  const stream = await media.getDisplayMedia({ video: true });
-  return new LocalDisplayTrack(stream);
+export async function createDisplayTrack(
+  constraints?: VideoConstraints
+): Promise<LocalDisplayTrack> {
+  const stream = await media.getDisplayMedia({ video: generateVideoConstraints(constraints) });
+  return new LocalDisplayTrack(stream.getVideoTracks()[0]);
+}
+
+/**
+ * Creates custom video track .
+ *
+ * @param config - Custom video track creation config.
+ * @returns A Promise that resolves to a LocalVideoTrack.
+ */
+export async function createCustomVideoTrack(
+  config: CustomVideoTrackInitConfig
+): Promise<LocalVideoTrack> {
+  const localVideoTrack = new LocalVideoTrack(config.mediaStreamTrack);
+
+  if (config.encoderConfig) {
+    localVideoTrack.setEncoderConfig(config.encoderConfig);
```

Comment:
where is `setEncoderConfig` defined? i don't see it in this PR and i don't see it in the `new-tracks` branch.

Reply:
its in a different PR let me consolidate them